romancal/source_detection/tests/test_source_detection_step.py

```diff
@@ -0,0 +1,252 @@
+"""
+ Unit tests for the Roman source detection step code
+"""
+
+import os
+
+import numpy as np
+import pytest
+from astropy import units as u
+from astropy.convolution import Gaussian2DKernel
+from roman_datamodels import maker_utils as testutil
+from roman_datamodels import units as ru
+from roman_datamodels.datamodels import ImageModel
+
+from romancal.lib import dqflags
+from romancal.source_detection import SourceDetectionStep
+
+
+@pytest.fixture
+def setup_inputs():
+    def _setup(nrows=100, ncols=100, noise=1.0):
+
+        """Return ImageModel of lvl 2 image"""
+
+        shape = (100, 100)  # size of test image
+        wfi_image = testutil.mk_level2_image(shape=shape)
+        wfi_image.data = u.Quantity(
+            np.ones(shape, dtype=np.float32), ru.electron / u.s, dtype=np.float32
+        )
+        wfi_image.meta.filename = "filename"
+
+        # add noise to data
+        if noise is not None:
+            wfi_image.data += u.Quantity(
+                noise * np.random.random(shape), ru.electron / u.s, dtype=np.float32
+            )
+
+        # add dq array
+
+        wfi_image.dq = np.zeros(shape, dtype=np.uint32)
+        # construct ImageModel
+        mod = ImageModel(wfi_image)
+
+        return mod
+
+    return _setup
+
+
+@pytest.mark.skipif(
+    os.environ.get("CI") == "true",
+    reason="Roman CRDS servers are not currently available outside the internal "
+    "network",
+)
+def add_random_gauss(arr, x_positions, y_positions, min_amp=200, max_amp=500):
+
+    """Add random 2D Gaussians to `arr` at specified positions,
+    with random amplitudes from `min_amp` to  `max_amp`. Assumes
+    units of e-/s."""
+
+    for i, x in enumerate(x_positions):
+        y = y_positions[i]
+        gauss = Gaussian2DKernel(2, x_size=20, y_size=20).array
+        amp = np.random.randint(200, 700)
+        arr[y - 10 : y + 10, x - 10 : x + 10] += (
+            u.Quantity(gauss, ru.electron / u.s, dtype=np.float32) * amp
+        )
```

Comment:
I think you'll have better centroid matching if you make this 21x21 instead of 20x20; even sized arrays get centered at a pixel corner rather than a pixel center.